### PR TITLE
added therapy and treament to repo and donor

### PR DESCRIPTION
**Pull request by jaserud**
**Description of changes**
- setup treament/therapy in documents and donor
- made changes to keep tests passing

Take a look, let me know what you think.

**Type of Change**

- [ ] Bug
- [x] Refactor
- [x] New Feature

**Checklist before requesting review:**

- [x] Check branch (code change PRs go to `develop` not master)
- [x] Manual testing
- [x] Regression tests completed and passing (double check number of tests).
- [x] Spelling has been checked.
- [ ] Updated swagger docs accordingly (check it's still valid)

### CODE REVIEW

File: src/clinical/clinical-entities.ts [hunk 1]
```diff
@@ -36,4 +34,18 @@ export interface Sample {
   submitterId: string;
 }
 
+export interface Treatment {
+  clinicalInfo?: ClinicalInfo;
+  chemotherapy: Array<Therapy>;
```
Comment (blabadi) [inline src/clinical/clinical-entities.ts]: the only difference between therapies is the name .. I'm thinking that we can generalize this .. a Treatment has a List of Therapies and each therapy has a name/type + clinical info

File: src/clinical/clinical-entities.ts [hunk 2]
```diff
@@ -36,4 +34,18 @@ export interface Sample {
...
+  hormoneTherapy: Array<Therapy>;
+}
+
+export interface Therapy {
+  clinicalInfo?: ClinicalInfo;
+}
+
+export interface ClinicalInfo {
+  [field: string]: string | number;
```
Comment (blabadi) [inline src/clinical/clinical-entities.ts]: this could also be boolean | undefined

Comment (jaserud) [reply src/clinical/clinical-entities.ts]: (:+1: ) true, I just copy pasta'd what was already their for all clinicalInfo, will update..

Comment (blabadi) [reply src/clinical/clinical-entities.ts]: yes but you also generalized this, before it was specific to donor and specimen, which according to the schema didn't have boolean or other stuff

File: src/clinical/clinical-entities.ts (hunk 1 above)
Comment (jaserud) [reply src/clinical/clinical-entities.ts]: hmm that's how I set it up at first, but then should `name/type` be constrain to an enum: ['hormoneTherapy', 'chemotherapy', ...] or is just `string` enough. I didn't want to define a SubClinicalEntity enum

File: src/clinical/clinical-entities.ts (hunk 2 above)
Comment (jaserud) [reply src/clinical/clinical-entities.ts]: do we want to keep that information or is it ok to generalize it?

Comment (blabadi) [reply src/clinical/clinical-entities.ts]: it's ok

File: src/clinical/clinical-entities.ts (hunk 1 above)
Comment (jaserud) [reply src/clinical/clinical-entities.ts]: ^ for that I mean in the mongoose document schema

Comment (blabadi) [reply src/clinical/clinical-entities.ts]: in mongoose  that can be a string

Comment (jaserud) [reply src/clinical/clinical-entities.ts]: cool, then I agree that makes more sense

**Pull request merged**